### PR TITLE
replace `ReactiveQueue` with `reactiveVal`

### LINKED ISSUE #163: Replace `ReactiveQueue` with simple `reactiveVal()`
`FilterStates` contains `ReactiveQueue` to store the `FIlterState`. This class also contains mechanism to add, remove, clear FilterState(s) but it's redundant as interaction with `ReactiveQueue` is done through `FilterStates` methods. 
Please remove `ReactiveQueue` from teal.slice and move `Queue` (and keep private) to the `teal.transform` where it's used.

### PR DESCRIPTION
**Pull request by chlebowa**
Closes #163.

Removes `Queue` and `ReactiveQueue` class (moved to `teal.transform`).
Rewrites `initialize` methods in all classes to use `reactiveVal` instead.
Rewrites queue methods accordingly.

Also some clean up in code and documentation.

### CODE REVIEW

**Review by gogonzo — COMMENTED** [review]
Looks good 👍

File: R/FilterStates.R [hunk 1]
```diff
@@ -184,34 +188,20 @@ FilterStates <- R6::R6Class( # nolint
...
+    #' @param x (`list` of `ReactiveQueue`)\cr
+    #'  must be a list even if single `ReactiveQueue` is set.
+    # queue_initialize = function(x) {
+    #   checkmate::assert_list(x, types = "ReactiveQueue", min.len = 1)
+    #   private$queue <- x
+    #   invisible(NULL)
+    # },
+    queue_initialize = function(x) {
+      checkmate::assert_list(x, min.len = 1)
+      private$queue <- x
```
Comment (nikolas-burkoff) [inline R/FilterStates.R:204]: I think this is confusing as it's not reactiveVal, but it is in all child classes. Maybe:
```suggestion
    queue_initialize = function(x) {
      checkmate::assert_list(x, min.len = 1)
      private$queue(x)
```
?

Comment (chlebowa) [reply R/FilterStates.R:204]: You mean change `private$queue` (renamed) in `FilterStates` from a list of `reactiveVal`s to a `reactiveVal` containing such a list?

Comment (nikolas-burkoff) [reply R/FilterStates.R:204]: ah - ok I misunderstood here I was confusing this function with the calls like this:

```
self$queue_initialize(
        list(
          y = reactiveVal()
        )
      )
```

I wonder if we actually need this `queue_initialize` function if we can just set it in the initializers directly

Comment (gogonzo) [reply R/FilterStates.R:204]: I support @nikolas-burkoff idea in this case

File: R/FilterStates.R [hunk 2]
```diff
@@ -116,7 +119,8 @@ FilterStates <- R6::R6Class( # nolint
         USE.NAMES = TRUE,
         simplify = FALSE,
         function(queue) {
-          items <- queue$get()
+          # items <- queue$get()
+          items <- queue()
```
[the remark this answers is not in the document]
Comment (chlebowa) [reply R/FilterStates.R:139]: renamed `queue` -> `state_list`
renamed `element_id` -> `state_id`

**Review by gogonzo — APPROVED** [review]
👍 Once checks are fix

**Comment by nikolas-burkoff** [comment]
Once merged into main please merge into filter_panel_refactor@main

File: NEWS.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 * Examples now use `scda.2022` instead of `scda.2021`.
 * Transferred data hashing step in `FilteredDataset` to `teal`.
+* Replaced `ReactiveQueue` objects with `shiny::reactiveVal` objects.
```
Comment (gogonzo) [inline NEWS.md:5]: Please don't bother with some internal change. FilterStates and ReactiveQueue were internal objects
```suggestion
```

Comment (chlebowa) [reply NEWS.md:5]: :+1:

**Comment by gogonzo** [comment]
@chlebowa hit the fire button 🧯

**Comment by chlebowa** [comment]
I am trying to prune `state_list_initialize` methods but I got a little lost.

**Pull request merged**